src/core/qgsvectordataprovider.h

```diff
@@ -246,6 +247,13 @@ class CORE_EXPORT QgsVectorDataProvider : public QgsDataProvider, public QgsFeat
      */
     virtual bool truncate();
 
+    /**
+     * Cancels the pending query.
```

Comment:
I don't really understand the intended use case for this call - what pending queries does a vector data provider have?

Reply:
For example with the `QgsVirtualLayerTask`, the `layer->reload()` blocking call runs the underlying sql query in a separated thread. Thanks to this method, we're able to cancel the query currently running.

But `pending query` is maybe badly named and confusing?

Reply:
Yes, I think the naming and docs need improving here. I think in this case it should be made quite specific, like "cancelReload"